draft-yasskin-dispatch-bundled-exchanges.md

```diff
@@ -0,0 +1,675 @@
+  SRI: W3C.REC-SRI-20160623
+  URL:
+    target: https://url.spec.whatwg.org/
+    title: URL
+    author:
+      org: WHATWG
+    date: Living Standard
+
+--- abstract
+
+Bundled exchanges provide a way to bundle up groups of HTTP request+response
+pairs to transmit or store them together. The component exchanges can be signed
+using {{?I-D.yasskin-http-origin-signed-responses}} to establish their
+authenticity.
+
+--- note_Note_to_Readers
+
+Discussion of this draft takes place on the ART area mailing list
+(art@ietf.org), which is archived
+at <https://mailarchive.ietf.org/arch/search/?email_list=art>.
+
+The source code and issues list for this draft can be found
+in <https://github.com/WICG/webpackage>.
+
+--- middle
+
+# Introduction
+
+To satisfy the use cases in {{?I-D.yasskin-webpackage-use-cases}}, this document
+proposes a new bundling format to group HTTP resources. Several of the use cases
+require the resources to be signed: that's provided by bundling signed exchanges
+({{?I-D.yasskin-http-origin-signed-responses}}) rather than natively in this
+format.
+
+## Terminology
+
+Exchange (noun)
+: An HTTP request/response pair. This can either be a request from a client and
+the matching response from a server or the request in a PUSH_PROMISE and its
+matching response stream. Defined by Section 8 of {{!RFC7540}}.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED",
+"MAY", and "OPTIONAL" in this document are to be interpreted as
+described in BCP 14 {{!RFC2119}} {{!RFC8174}} when, and only when, they
+appear in all capitals, as shown here.
+
+## Mode of specification {#mode}
+
+This specification defines how conformant bundle parsers work. It does not
+constrain how encoders produce a bundle: although there are some guidelines in
+{{authoring-guidelines}}, encoders MAY produce any sequence of bytes that a
+conformant parser would parse into the intended semantics.
+
+This specification uses the conventions and terminology defined in the Infra
+Standard ({{INFRA}}).
+
+# Semantics {#semantics}
+
+A bundle is logically a set of HTTP exchanges, with a URL identifying the
+manifest(s) of the bundle itself. A client uses content negotiation to pick a
+particular response for that URL, often an App Manifest ({{appmanifest}}), which
+then identifies other significant exchanges within the bundle, like a start
+page.
+
+While the order of the exchanges is not semantically meaningful, it can
+significantly affect performance when the bundle is loaded from a network
+stream.
+
+A bundle is parsed from a stream of bytes, which is assumed to have the attributes and operations described in {{stream-operations}}:
+
+Bundle parsers support two operations, defined in {{semantics-load-metadata}}
+and {{semantics-load-response}} each of which can return an error instead of
+their normal result.
+
+## Stream attributes and operations {#stream-operations}
```

Comment:
@domenic, [here's the use of streams](https://jyasskin.github.io/webpackage/bundles/draft-yasskin-dispatch-bundled-exchanges.html#stream-operations) I mentioned on #whatwg. I'm thinking about re-doing it in terms of [`ReadableStream`](https://fetch.spec.whatwg.org/#readablestream) despite your advice because https://jyasskin.github.io/webpackage/bundles/draft-yasskin-dispatch-bundled-exchanges.html#load-response needs to create a [body's stream](https://fetch.spec.whatwg.org/#concept-body), which is defined as a `ReadableStream`. The stream a bundle's parsed from is likely to be a response body anyway, so it'll probably be a ReadableStream even if I pretend it isn't.